tests/src/tools/ReadyToRun.TestHarness/Program.cs

```diff
@@ -33,82 +34,83 @@ class Program
+                return 0;
             }
 
-            if (!File.Exists(args[0]))
+            if (!File.Exists(_coreRunExePath))
             {
-                Console.WriteLine($"Error: {args[0]} is an invalid path.");
+                Console.WriteLine($"Error: {_coreRunExePath} is an invalid path.");
                 ShowUsage();
                 return exitCode;
             }
 
-            if (!File.Exists(args[1]))
+            if (!File.Exists(_testExe))
             {
-                Console.WriteLine($"Error: {args[1]} is an invalid path.");
+                Console.WriteLine($"Error: {_testExe} is an invalid path.");
                 ShowUsage();
                 return exitCode;
             }
 
-            int passThroughIndex = -1;
             string passThroughArguments = "";
-            if (args.Length >= 3)
+            if (_testargs.Count > 0)
             {
-                if (args[2] == "/testargs")
-                {
-                    passThroughIndex = 2;
-                } else
-                {
-                    if (!File.Exists(args[2]))
-                    {
-                        Console.WriteLine($"Error: {args[2]} is an invalid path.");
-                        ShowUsage();
-                        return exitCode;
-                    }
-                    whiteListFile = args[2];
-
-                    if (args.Length >= 4 && args[3] == "/testargs")
-                    {
-                        passThroughIndex = 3;
-                    }
-                }
-                
-                if (passThroughIndex > -1 && (args.Length - passThroughIndex - 1) > 0)
-                {
-                    passThroughArguments = string.Join(' ', args, passThroughIndex + 1, args.Length - passThroughIndex - 1);
-                }
+                passThroughArguments = string.Join(' ', _testargs);
             }
 
-            string coreRunExePath = args[0];
-            string testExe = args[1];
-
-            // TODO: CoreCLR test bed has tests with multiple assemblies - we'll need to add them here when we support that
             var testModules = new HashSet<string>();
-            testModules.Add(testExe);
+            testModules.Add(_testExe);
+            foreach (string reference in _referenceFilenames)
+            {
+                testModules.Add(reference);
+            }
 
             using (var session = new TraceEventSession("ReadyToRunTestSession"))
             {
                 var r2rMethodFilter = new ReadyToRunJittedMethods(session, testModules);
                 session.EnableProvider(ClrTraceEventParser.ProviderGuid, TraceEventLevel.Verbose, (ulong)(ClrTraceEventParser.Keywords.Jit | ClrTraceEventParser.Keywords.Loader));
                 
-                Task.Run(() => exitCode = RunTest(session, coreRunExePath, testExe, passThroughArguments));
-
-                // Block, processing callbacks for events we subscribed to
-                session.Source.Process();
```

Comment:
Was this accidentally removed? This line is needed to ensure events are processed.